airflow/executors/dask_executor.py

```diff
@@ -44,6 +42,7 @@ def __init__(self, cluster_address=None):
         super().__init__(parallelism=0)
 
     def start(self):
+        import distributed
```

Comment:
I think we should avoid the whole file if it's not needed. This causes the problems with importing non-existing libraries to disappear.

Reply:
What do you mean? The 'distributed' package is added as dependency when 'dask' is installed. It is not installed even when "All" is configured. I'd love to get rid of dask executor - I think no-one uses it any more but this is a process that requires voting and community to agree. Right now I try to get rid of cyclic import problems and not get rid of dask executor.

Do you have a suggestion what else can we do now?

Reply:
After introduction suggestions from comment - https://github.com/apache/airflow/pull/6596/files#r347158027 , the problem will disappear. The dask_executor.py file will not be loaded if the executor is not selected, so the `distributed` library is probably not installed.

Reply:
The changes to the airflow/executors/* shouldn't be needed now, right?

Reply:
I think it is still nicer to have it this way. I think it is much better to have only one place from which you should import something. Especially if you have some logic in __init__() files, just importing classes to init from different packages so that you can have shorter import path (what else) is not a good pattern I think and it's better to untangle it now - while we are at this stage of 2.0 - with executors I think we are pretty safe to remove them from executors.__init__ because no-one was supposed to use them differently. I don't even we need to have any deprecation there, it's pretty much internal for Airflow. And since we have monorepo, we can easily change it. 

I can make an extra note in UPDATING.md about it though.

Reply:
(I am talking here about removing import LocalExecutor/import SequentialExecutor in the __init__. This was really bad pattern. I will bring the "AvailableExecutors" class back to __init__ however.

Reply:
Yeah delaying the import of `*Executor` until needed def makes sense to stay.

But by delaying the use of this import possibly delays errors due to missing packages for the current/selected executor untli task execution, instead of "at start up"

Reply:
Actually from what I see it's not wene task is started but when the default executor is instantiated in the _run() method of the CLI which is exactly where we want it to happen.

Reply:
And then only the deps needed by that given executor will be loaded.